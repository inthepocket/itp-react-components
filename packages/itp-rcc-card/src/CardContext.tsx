import { createContext } from 'react';
import CardStylesInterface from './CardStylesInterface';

interface CardContextInterface {
  prefixCss?: string | null;
  size: 'small' | 'default' | 'large';
  styles?: CardStylesInterface | null;
}

/**
 * Card Context
 */
const CardContext =
  createContext<CardContextInterface>({ prefixCss: 'card', size: 'default', styles: null });

export default CardContext;
