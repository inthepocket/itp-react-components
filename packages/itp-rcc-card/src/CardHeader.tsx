import * as React from 'react';
import capitalize from 'capitalize';
import classnames from 'classnames';
import CardContext from './CardContext';
import CardChildPropsInterface from './CardChildPropsInterface';

const { Consumer } = CardContext;

/**
 * Card Header
 */
const CardHeader = ({ children }: CardChildPropsInterface) => {
  return (
    <Consumer>
      {(cardContext) => {
        const { prefixCss, size, styles } = cardContext;
        const containerClassName = styles ? styles.header : `${prefixCss}-header`;
        const sizeClassName = styles ? styles[`headerSize${capitalize(size)}`] : `${prefixCss}-header-size-${size}`;

        const className = classnames({
          [`${containerClassName}`]: containerClassName,
          [`${sizeClassName}`]: sizeClassName,
        });

        return (
          <header className={className}>{children}</header>
        );
      }}
    </Consumer>
  );
};

export default CardHeader;
