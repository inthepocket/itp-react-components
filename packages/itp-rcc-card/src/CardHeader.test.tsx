import * as React from 'react';
import { create } from 'react-test-renderer';
import CardHeader from './CardHeader';
import CardContext from './CardContext';

const { Provider } = CardContext;
const cssModuleMock = {
  body: 'cssModulesBody',
  header: 'cssModulesHeader',
  headerSizeSmall: 'cssModulesHeaderSizeSmall',
  headerSizeLarge: 'cssModulesHeaderSizeLarge',
  card: 'cssModulesCard',
};

test('CardHeader: default', () => {
  const tree = create((
    <CardHeader>
      <p>Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.</p>
    </CardHeader>
  ))
  .toJSON();

  expect(tree).toMatchSnapshot();
});

test('CardHeader: size: small', () => {
  const tree = create((
    <Provider value={{ prefixCss: 'card', size: 'small' }}>
      <CardHeader>
        <p>Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.</p>
      </CardHeader>
    </Provider>
  ))
  .toJSON();

  expect(tree).toMatchSnapshot();
});

test('CardHeader: size: large', () => {
  const tree = create((
    <Provider value={{ prefixCss: 'card', size: 'large' }}>
      <CardHeader>
        <p>Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.</p>
      </CardHeader>
    </Provider>
  ))
  .toJSON();

  expect(tree).toMatchSnapshot();
});

test('CardHeader: css modules: default', () => {
  const tree = create((
    <Provider value={{ prefixCss: 'card', size: 'default', styles: cssModuleMock }}>
      <CardHeader>
        <p>Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.</p>
      </CardHeader>
    </Provider>
  ))
  .toJSON();

  expect(tree).toMatchSnapshot();
});

test('CardHeader: css modules: size: small', () => {
  const tree = create((
    <Provider value={{ prefixCss: 'card', size: 'small', styles: cssModuleMock }}>
      <CardHeader>
        <p>Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.</p>
      </CardHeader>
    </Provider>
  ))
  .toJSON();

  expect(tree).toMatchSnapshot();
});

test('CardHeader: css modules: size: large', () => {
  const tree = create((
    <Provider value={{ prefixCss: 'card', size: 'large', styles: cssModuleMock }}>
      <CardHeader>
        <p>Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.</p>
      </CardHeader>
    </Provider>
  ))
  .toJSON();

  expect(tree).toMatchSnapshot();
});