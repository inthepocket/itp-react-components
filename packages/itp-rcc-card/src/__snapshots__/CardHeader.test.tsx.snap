// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardHeader: css modules: default 1`] = `
<header
  className="cssModulesHeader"
>
  <p>
    Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.
  </p>
</header>
`;

exports[`CardHeader: css modules: size: large 1`] = `
<header
  className="cssModulesHeader cssModulesHeaderSizeLarge"
>
  <p>
    Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.
  </p>
</header>
`;

exports[`CardHeader: css modules: size: small 1`] = `
<header
  className="cssModulesHeader cssModulesHeaderSizeSmall"
>
  <p>
    Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.
  </p>
</header>
`;

exports[`CardHeader: default 1`] = `
<header
  className="card-header card-header-size-default"
>
  <p>
    Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.
  </p>
</header>
`;

exports[`CardHeader: size: large 1`] = `
<header
  className="card-header card-header-size-large"
>
  <p>
    Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.
  </p>
</header>
`;

exports[`CardHeader: size: small 1`] = `
<header
  className="card-header card-header-size-small"
>
  <p>
    Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.
  </p>
</header>
`;
